### PR TITLE
shorten proof of sacgr

### PR DESCRIPTION
**Pull request by expln**

### CODE REVIEW

**Comment by wlammen** [comment]
Thanks for your contribution.

The new proof roughly eliminates lines 26 - 37 of the old proof.  No additional axioms are introduced, no extra definitions are used.  Without knowing the topic and techniques in detail, this shortening looks ok.
There are formal objections:
No shortening tag is added, the OLD version is not kept.  See below a suggestion how it could be done.   Enter your name in the suggested location.  Afterwards, you need to recreate the 'discouraged' file and add it to your commitment, by

`./metamath 'read set.mm' 'set width 9999' 'show discouraged' quit | tr -d '\015' | grep '^SHOW DISCOURAGED.' | sed -E -e 's/^SHOW DISCOURAGED.  ?//' | LC_ALL=C sort > discouraged
` 

>
    ${
      $d X x y $.  $d Y x y $.
      sacgr.x $e |- ( ph -> X e. P ) $.
      sacgr.y $e |- ( ph -> Y e. P ) $.
      sacgr.1 $e |- ( ph -> <" A B C "> ( cgrA ` G ) <" D E F "> ) $.
      sacgr.2 $e |- ( ph -> B e. ( A I X ) ) $.
      sacgr.3 $e |- ( ph -> E e. ( D I Y ) ) $.
      sacgr.4 $e |- ( ph -> B =/= X ) $.
      sacgr.5 $e |- ( ph -> E =/= Y ) $.
      $( Supplementary angles of congruent angles are themselves congruent.
         Theorem 11.13 of [Schwabhauser] p. 98.  (Contributed by Thierry
         Arnoux, 30-Sep-2020.)    (Proof shortened by <PUT YOUR NAME HERE>,
         16-Feb-2023.) $)
      sacgr $p |- ( ph -> <" X B C "> ( cgrA ` G ) <" Y E F "> ) $=
        ( vx vy cmir cfv cs3 cv ccgrg wbr chlg w3a ccgra wcel wa eqid ad3antrrr
        cstrkg clng simpllr mircl simplr mirmir eqidd s3eqd necomd mirne simpr1
        wceq mirtrcgr eqbrtrrd simpr2 hlne1 co hlcomd btwnhl tgbtwncom eleqtrrd
        oveq2d mirhl2 simpr3 iscgrad wrex cgrane2 cgraid cgrane1 cgrahl1 cgratr
        wne iscgra mpbid r19.29vva ) ALCIUMUNZUNZUNZCDUOZUKUPZGULUPZUOIUQUNZURZ
        XEEGIUSUNZUNZURZXFHXJURZUTZLCDUOZMGHUOIVAUNZURUKULFFAXEFVBZVCZXFFVBZVCZ
        XMVCZLCDMFGHIJXIXEGXAUNZUNZXFNOXIVDZAIVFZVBXPXRXMQVEZALFVBXPXRXMUDVEACF
        VBXPXRXMSVEZADFVBXPXRXMTVEZAMFVBXPXRXMUEVEZAGFVBXPXRXMUBVEZAHFVBXPXRXMU
        CVEXTGFXAIJIVGUNZYAKXENPOYJVDZXAVDZYEYIYAVDZAXPXRXMVHZVIZXQXRXMVJZXTXCX
        BUNZCDUOZXNYBGXFUOXGAYRXNVQXPXRXMAYQCDDLCACLFXAIJYJXBKNPOYKYLQSXBVDZUDV
        KZACVLADVLVMVEXTXCCDFXGXAIJYJXBKYAXEGXFNPOYKYLYEXGVDYSYMAXCFVBXPXRXMACF
        XAIJYJXBKLNPOYKYLQSYSUDVIZVEYFYNYIYGYPAXCCWQXPXRXMACLFXAIJYJXBKNPOYKYLQ
        SYSUDACLUIVNVOZVEXSXHXKXLVPVRVSXTMYBGFIJXIYDNOYCYHYOYIYEXTGFXAIJXIYJYAK
        MYBGNPOYKYLYEYMYCYIYHYOYIXTGMAGMWQXPXRXMUJVEVNXTGXEFXAIJYJYAKNPOYKYLYEY
        IYMYNXTXEEGFIJXIYDNOYCYNAEFVBXPXRXMUAVEZYIYEXSXHXKXLVTZWAVOXTGMXEJWBMYB
        YAUNZJWBXTXEGMFIJKNPOYEYNYIYHXTEXEMGFIJXINOYCUUCYNYHYEYIXTXEEGFIJXIYDNO
        YCYNUUCYIYEUUDWCAGEMJWBVBXPXRXMUHVEWDWEXTUUEXEMJXTGXEFXAIJYJYAKNPOYKYLY
        EYIYMYNVKWGWFWHWCXSXHXKXLWIWJAXDEGHUOXOURXMULFWKUKFWKAXCCDBFGCDIEJHXINO
        QYCUUASTRSTAXCCDXCFCDIJXIBNOYCQUUASTUUASTAXCCDFIJXINOQYCUUASTUUBABCDEFG
        HIJXINOYCQRSTUAUBUCUFWLWMRACFXAIJXIYJXBKBXCBNPOYKYLQYSYCSRUUARABCDEFGHI
        JXINOYCQRSTUAUBUCUFWNUUBACBLJWBBYQJWBUGAYQLBJYTWGWFWHWOUAUBUCUFWPAUKULX
        CCDEFGHIJXINOYCQUUASTUAUBUCWRWSWT $.

      $( Obsolete version of ~ sacgr as of 16-Feb-2023.  (Contributed by
         Thierry Arnoux, 30-Sep-2020.)  (Proof modification is discouraged.)
         (New usage is discouraged.) $)
      sacgrOLD $p |- ( ph -> <" X B C "> ( cgrA ` G ) <" Y E F "> ) $=
        ( vx vy cmir cfv cs3 cv ccgrg wbr chlg w3a ccgra wcel wa eqid ad3antrrr
        cstrkg simpllr mircl simplr mircgr tgcgrcomlr simpr1 cgr3simp1 3eqtr2rd
        clng co cgr3simp2 mirmir eqidd s3eqd wne necomd mirne mirtrcgr eqbrtrrd
        cgr3swap13 cgr3simp3 trgcgr simpr2 hlne1 hlcomd btwnhl tgbtwncom oveq2d
        wceq eleqtrrd mirhl2 simpr3 iscgrad wrex cgrane2 cgraid cgrane1 cgrahl1
        cgratr iscgra mpbid r19.29vva ) ALCIUMUNZUNZUNZCDUOZUKUPZGULUPZUOIUQUNZ
        URZXMEGIUSUNZUNZURZXNHXRURZUTZLCDUOZMGHUOIVAUNZURUKULFFAXMFVBZVCZXNFVBZ
        VCZYAVCZLCDMFGHIJXQXMGXIUNZUNZXNNOXQVDZAIVFVBYDYFYAQVEZALFVBYDYFYAUDVEZ
        ACFVBYDYFYASVEZADFVBYDYFYATVEZAMFVBYDYFYAUEVEZAGFVBYDYFYAUBVEZAHFVBYDYF
        YAUCVEYHGFXIIJIVOUNZYIKXMNPOYRVDZXIVDZYLYQYIVDZAYDYFYAVGZVHZYEYFYAVIZYH
        LCDYJFXOGXNIKNPXOVDZYLYMYNYOUUCYQUUDYHYJGKVPXMGKVPXKCKVPZLCKVPZYHGYJGXM
        FIJKNPOYLYQUUCYQUUBYHGXMFXIIJYRYIKNPOYSYTYLYQUUAUUBVJVKYHXKCDXMFXOGXNIJ
        KNPOUUEYLAXKFVBYDYFYAACFXIIJYRXJKLNPOYSYTQSXJVDZUDVHZVEZYNYOUUBYQUUDYGX
        PXSXTVLZVMAUUFUUGWOYDYFYAACXKCLFIJKNPOQSUUISUDACLFXIIJYRXJKNPOYSYTQSUUH
        UDVJVKVEVNYHXKCDXMFXOGXNIJKNPOUUEYLUUJYNYOUUBYQUUDUUKVQYHLDYJXNFIJKNPOY
        LYMYOUUCUUDYHDCLXNFXOGYJIJKNPOUUEYLYOYNYMUUDYQUUCYHLCDYJFXOGXNIJKNPOUUE
        YLYMYNYOUUCYQUUDYHXKXJUNZCDUOZYBYJGXNUOXOAUUMYBWOYDYFYAAUULCDDLCACLFXII
        JYRXJKNPOYSYTQSUUHUDVRZACVSADVSVTVEYHXKCDFXOXIIJYRXJKYIXMGXNNPOYSYTYLUU
        EUUHUUAUUJYNUUBYQYOUUDAXKCWAYDYFYAACLFXIIJYRXJKNPOYSYTQSUUHUDACLUIWBWCZ
        VEUUKWDWEWFWGVKWHYHMYJGFIJXQVFNOYKYPUUCYQYLYHGFXIIJXQYRYIKMYJGNPOYSYTYL
        UUAYKYQYPUUCYQYHGMAGMWAYDYFYAUJVEWBYHGXMFXIIJYRYIKNPOYSYTYLYQUUAUUBYHXM
        EGFIJXQVFNOYKUUBAEFVBYDYFYAUAVEZYQYLYGXPXSXTWIZWJWCYHGMXMJVPMYJYIUNZJVP
        YHXMGMFIJKNPOYLUUBYQYPYHEXMMGFIJXQNOYKUUPUUBYPYLYQYHXMEGFIJXQVFNOYKUUBU
        UPYQYLUUQWKAGEMJVPVBYDYFYAUHVEWLWMYHUURXMMJYHGXMFXIIJYRYIKNPOYSYTYLYQUU
        AUUBVRWNWPWQWKYGXPXSXTWRWSAXLEGHUOYCURYAULFWTUKFWTAXKCDBFGCDIEJHXQNOQYK
        UUISTRSTAXKCDXKFCDIJXQBNOYKQUUISTUUISTAXKCDFIJXQNOQYKUUISTUUOABCDEFGHIJ
        XQNOYKQRSTUAUBUCUFXAXBRACFXIIJXQYRXJKBXKBNPOYSYTQUUHYKSRUUIRABCDEFGHIJX
        QNOYKQRSTUAUBUCUFXCUUOACBLJVPBUULJVPUGAUULLBJUUNWNWPWQXDUAUBUCUFXEAUKUL
        XKCDEFGHIJXQNOYKQUUISTUAUBUCXFXGXH $.
    $}

File: set.mm [hunk 1]
```diff
@@ -389459,34 +389459,27 @@ then any point of the half line ( ` R ` ` A ` ) also lies opposite to
          Arnoux, 30-Sep-2020.) $)
```
Comment (wlammen) [inline set.mm:389459]: Add your name in a 'Proof shortened' tag here.

File: set.mm [hunk 2]
```diff
@@ -389459,34 +389459,27 @@ then any point of the half line ( ` R ` ` A ` ) also lies opposite to
...
+        MYBGNPOYKYLYEYMYCYIYHYOYIXTGMAGMWQXPXRXMUJVEVNXTGXEFXAIJYJYAKNPOYKYLYEY
+        IYMYNXTXEEGFIJXIYDNOYCYNAEFVBXPXRXMUAVEZYIYEXSXHXKXLVTZWAVOXTGMXEJWBMYB
+        YAUNZJWBXTXEGMFIJKNPOYEYNYIYHXTEXEMGFIJXINOYCUUCYNYHYEYIXTXEEGFIJXIYDNO
+        YCYNUUCYIYEUUDWCAGEMJWBVBXPXRXMUHVEWDWEXTUUEXEMJXTGXEFXAIJYJYAKNPOYKYLY
+        EYIYMYNVKWGWFWHWCXSXHXKXLWIWJAXDEGHUOXOURXMULFWKUKFWKAXCCDBFGCDIEJHXINO
+        QYCUUASTRSTAXCCDXCFCDIJXIBNOYCQUUASTUUASTAXCCDFIJXINOQYCUUASTUUBABCDEFG
+        HIJXINOYCQRSTUAUBUCUFWLWMRACFXAIJXIYJXBKBXCBNPOYKYLQYSYCSRUUARABCDEFGHI
+        JXINOYCQRSTUAUBUCUFWNUUBACBLJWBBYQJWBUGAYQLBJYTWGWFWHWOUAUBUCUFWPAUKULX
+        CCDEFGHIJXINOYCQUUASTUAUBUCWRWSWT $.
     $}
```
Comment (wlammen) [inline set.mm:389483]: keep the old proof as sacgrOLD.  For details see my comment in the conversations part.

**Review by wlammen — CHANGES_REQUESTED** [review]

**Comment by avekens** [comment]
After @wlammen`s remarks are respected, this PR will be fine for me. Maybe @tirix (as contributor of the theorem) should also have a look at it.

**Comment by tirix** [comment]
This shortens the proof a lot, thank you for spotting this @expln !

I reviewed the proof and I'm OK with the shortening:
The new proof still follows Schwabhäuser's reference. 
It turns out that I took several steps to come back to the same statement within the proof. The new proof takes the obvious shortcut and removes the extraneous steps. 

I suppose that it has been obtained automatically? It's not explained in [the mail on the group](https://groups.google.com/g/metamath/c/B9_PEy8YjU4/m/JCJPdz_RAQAJ?utm_medium=email&utm_source=footer), but I'd be interested to know the process. Especially, is that something we can run periodically like `minimize`, or even within the checks?

**Review by tirix — APPROVED** [review]
The changes are OK with me, provided Wolf's remark about the name is fixed.
(If you don't want to provide your real name, `expln` would be fine too)

I don't think it's mandatory to keep the old proof in this case, however it is the custom.

**Comment by benjub** [comment]
> If you don't want to provide your real name, `expln` would be fine too

Initials are nice too ! (and there is a "key" at the top of set.mm for contributors)

**Comment by expln** [comment]
Thanks for the review, I will do all the required changes in a day or two.

**Comment by expln** [comment]
I made all the required changes. Please review again.

**Review by wlammen — APPROVED** [review]

**Pull request merged**